src/loaders/agent-base.js

```diff
@@ -4,14 +4,15 @@ import { warn } from '../common/util/console'
 import { SR_EVENT_EMITTER_TYPES } from '../features/session_replay/constants'
 import { generateRandomHexString } from '../common/ids/unique-id'
 import { TimeKeeper } from '../common/timing/time-keeper'
+import { globalScope } from '../common/constants/runtime'
 
 /**
  * @typedef {import('./api/interaction-types').InteractionInstance} InteractionInstance
  */
 
 export class AgentBase {
   agentIdentifier
-  timeKeeper = new TimeKeeper(this)
+  timeKeeper = new TimeKeeper(Math.floor(globalScope?.performance?.timeOrigin || globalScope?.performance?.timing?.navigationStart || Date.now()))
```

Comment:
This is fine but it represents the third place we've defined the origin time so we just need to follow up and remove `runtime.offset` and `runtime/constants.offset` at some time

Reply:
`timing/now` as well

Reply:
Thats the plan.